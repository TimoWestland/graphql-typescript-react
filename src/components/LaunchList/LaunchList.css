.launch-list {
  height: 100vh;
  overflow: hidden auto;
  background-color: #818181;
  width: 300px;
  padding-left: 20px;
  padding-right: 20px;
}

.launch-list__list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.launch-list__item {
  padding-top: 20px;
  padding-bottom: 20px;
  border-top: 1px solid #919191;
  cursor: pointer;
}
