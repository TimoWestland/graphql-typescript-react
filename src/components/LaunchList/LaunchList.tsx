import './LaunchList.css'

import * as React from 'react'
import { LaunchListQuery } from '../../generated/graphql'

export interface OwnProps {
  onIdChange: (newId: number) => void
}

export interface Props extends OwnProps {
  data: LaunchListQuery
}

export const LaunchList: React.FC<Props> = ({ data, onIdChange }) => (
  <div className="launch-list">
    <h3>Launches</h3>
    <ol className="launch-list__list">
      {!!data.launches &&
        data.launches.map(
          (launch, i) =>
            !!launch && (
              <li
                key={i}
                data-testid="launch-list-item"
                className="launch-list__item"
                onClick={() => onIdChange(launch.flight_number!)}
              >
                {launch.mission_name} {launch.launch_year}
              </li>
            )
        )}
    </ol>
  </div>
)

export default LaunchList
