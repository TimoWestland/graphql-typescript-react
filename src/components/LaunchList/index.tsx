import * as React from 'react'
import { useLaunchListQuery } from '../../generated/graphql'
import LaunchList, { OwnProps } from './LaunchList'

const LaunchListContainer: React.FC<OwnProps> = (props) => {
  const { data, error, loading } = useLaunchListQuery()

  if (loading) {
    return <div>Loading...</div>
  }

  if (error || !data) {
    return <div>Something went wrong!</div>
  }

  return <LaunchList data={data} {...props} />
}

export default LaunchListContainer
