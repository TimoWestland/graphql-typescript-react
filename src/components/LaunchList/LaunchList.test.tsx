import * as React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import LaunchList, { Props } from './LaunchList'

const mockData = {
  launches: [
    {
      flight_number: 1,
      mission_name: 'some name',
      launch_year: 2019,
    },
    {
      flight_number: 2,
      mission_name: 'some name',
      launch_year: 2019,
    },
    {
      flight_number: 3,
      mission_name: 'some name',
      launch_year: 2019,
    },
  ],
}

describe('LaunchList', () => {
  const onIdChange = jest.fn()
  const defaultProps: Props = {
    onIdChange,
    data: mockData,
  }

  const create = (props: Partial<Props> = {}) => render(<LaunchList {...defaultProps} {...props} />)

  afterEach(() => {
    cleanup()
    onIdChange.mockReset()
  })

  it('should match snapshot', () => {
    const { container } = create()
    expect(container).toMatchSnapshot()
  })

  it('should handle id change', () => {
    const { getAllByTestId } = create()
    const items = getAllByTestId('launch-list-item')

    fireEvent.click(items[0])
    expect(defaultProps.onIdChange).toHaveBeenCalledWith(1)

    fireEvent.click(items[1])
    expect(defaultProps.onIdChange).toHaveBeenCalledWith(2)
  })
})
