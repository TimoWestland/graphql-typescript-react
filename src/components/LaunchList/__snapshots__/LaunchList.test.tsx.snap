// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchList should match snapshot 1`] = `
<div>
  <div
    class="launch-list"
  >
    <h3>
      Launches
    </h3>
    <ol
      class="launch-list__list"
    >
      <li
        class="launch-list__item"
        data-testid="launch-list-item"
      >
        some name
         
        2019
      </li>
      <li
        class="launch-list__item"
        data-testid="launch-list-item"
      >
        some name
         
        2019
      </li>
      <li
        class="launch-list__item"
        data-testid="launch-list-item"
      >
        some name
         
        2019
      </li>
    </ol>
  </div>
</div>
`;
